### PR TITLE
rgw: use pre-defined calls to replace raw flag operation

### PR DESCRIPTION
**Pull request by guihecheng**
Signed-off-by: Gui Hecheng <guihecheng@cmss.chinamobile.com>

### CODE REVIEW

File: src/rgw/rgw_file.h [hunk 1]
```diff
@@ -506,7 +506,7 @@ namespace rgw {
 
     int open(uint32_t gsh_flags) {
       lock_guard guard(mtx);
-      if (! (flags & FLAG_OPEN)) {
+      if (! (is_open())) {
```
Comment (joscollin) [inline src/rgw/rgw_file.h]: Can you remove the unnecessary brackets ?

Comment (mattbenjamin) [inline src/rgw/rgw_file.h]: @joscollin ++

**Comment by mattbenjamin** [comment]
@guihecheng agree with above nit, but agree these consolidations are fine

File: src/rgw/rgw_file.h (hunk 1 above)
Comment (guihecheng) [inline src/rgw/rgw_file.h]: Oh, yes, thanks.

**Comment by guihecheng** [comment]
@mattbenjamin @joscollin updated as suggested :)

**Pull request merged**